### DISCUSSION TITLE
这里是读者对初稿的反馈意见

### DISCUSSION THREAD

**Discussion by yyc-git** [yyc-git/3DProgramPattern#2] [Announcements]
欢迎各位读者在这里提出您对初稿的反馈意见，让这本书变得更好，感谢您！

您也可以发起PR，在下面几个方面进行改进：
- 修改明显的错误
- 在哪个地方认为需要进行修改的，可以在上面通过加上注释来给出修改意见

**Comment by Aluomolo** [comment]
模式章节代码量占比过大，读者阅读困难，初学者劝退。

**Reply by yyc-git** [reply]
谢谢反馈，后面会对代码进行抽象，用伪代码替代源码

**Comment by jinleili** [comment]
ECS 模式阅读反馈 :)
首先声明，作为 ECS 模式的初学者，我目前仅粗浅地使用过基于 ECS 模式的 Bevy 引擎及阅读过《Hands-on Rust: Effective Learning through 2D Game Development and Play》(一本 Rust + ECS 模式开发游戏的入门书籍)，所以我的阅读感受肯定是受限于自己对 ECS 的。

文章优势：不依赖任何第三方 ECS 库，直观地展示了由面向对象到 ECS 模式的演化及具体实现。
文章劣势：正是由于展示了 ECS 本身的内部实现细节，像我这类不关注 ECS 内部细节，只是想使用 ECS 的读者而言，这些细节可能会掩盖 ECS 模式的优势。

**Reply by yyc-git** [reply]
谢谢

**Comment by l1305599284** [comment]
如果主要讲设计的话，需要避免一些技术上的专业性，减少代码量，深入浅出
可以以一个更通俗的角度去讲，降低用户门槛

**Reply by yyc-git** [reply]
谢谢
